src/server.ts

```diff
@@ -26,120 +33,170 @@ export class MyAgent extends Agent<Env> {
     apiKey: this.env.GROQ_API_KEY,
   })
   messages = RESET
+  connection!: Connection
 
   constructor(state: DurableObjectState, env: Env) {
     super(state, env)
   }
 
   onConnect(connection: Connection, ctx: ConnectionContext): void | Promise<void> {
+    this.connection = connection
     console.log('Connected to agent')
     // Clear history
     this.messages = RESET
-    this.respond(connection, {
+    this.respond({
       role: 'assistant',
       content: `Hello! I'm your AI assistant. How can I help you today?`,
     })
   }
 
-  respond(connection: Connection, message: CoreAssistantMessage | CoreToolMessage) {
+  respond(message: CoreAssistantMessage | CoreToolMessage) {
```

Comment:
This whole method is just me learning how the fuck tool call responses work. Please tell me the Agent framework is going to make all this just disappear?